Code/GraphMol/MolPickler.cpp

```diff
@@ -408,6 +428,9 @@ Query<int, T const *, true> *buildBaseQuery(std::istream &ss, T const *owner,
       static_cast<RangeQuery<int, T const *, true> *>(res)->setUpper(val);
       streamRead(ss, val, version);
       static_cast<RangeQuery<int, T const *, true> *>(res)->setTol(val);
+      streamRead(ss, cval, version);
+      static_cast<RangeQuery<int, T const *, true> *>(res)->setEndsOpen(
+          cval & (1 << 1), cval & 1);
```

Comment:
Why use the shift op here?  It might be cleaner as (cval & 0x2).

Reply:
I see, it's a mirror of the pickler.

Reply:
I see, it's a mirror of the pickler
```
    ends = 0 | (rdcast<int>(lowerOpen) << 1) | rdcast<int>(upperOpen);
```

Still, might be easier as the following:

```
const unsigned int LowerOpen = (1<<1);
const unsigned int UpperOpen = 1;
...
static_cast<RangeQuery<int, T const *, true> *>(res)->setEndsOpen(cval & LowerOpen, cval & UpperOpen);
          
```